backend/internal/service/version_service.go

```diff
@@ -0,0 +1,80 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync/atomic"
+	"time"
+)
+
+const versionCacheTTL = 15 * time.Minute
+
+type cacheEntry struct {
+	version     string
+	lastFetched time.Time
+}
+
+type VersionService struct {
+	httpClient *http.Client
+	cache      atomic.Value
+}
+
+func NewVersionService(httpClient *http.Client) *VersionService {
+	s := &VersionService{httpClient: httpClient}
+	s.cache.Store((*cacheEntry)(nil))
+	return s
+}
+
+// GetLatestVersion returns the latest available version from GitHub.
+// It caches the result for a short duration to avoid excessive API calls.
+func (s *VersionService) GetLatestVersion(ctx context.Context) (string, error) {
+	// Serve from cache if fresh
+	if entry, ok := s.cache.Load().(*cacheEntry); ok && entry != nil {
+		if time.Since(entry.lastFetched) < versionCacheTTL {
+			return entry.version, nil
+		}
+	}
+
+	// Fetch from GitHub
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet,
+		"https://api.github.com/repos/pocket-id/pocket-id/releases/latest", nil)
+	if err != nil {
+		return "", fmt.Errorf("create GitHub request: %w", err)
+	}
+
+	httpClient := *s.httpClient
+	httpClient.Timeout = 2 * time.Second
+
+	resp, err := httpClient.Do(req)
```

Comment:
Avoiding creating a new HTTP client just to add a timeout:

```suggestion
	reqCtx, cancel := context.WithTimeout(ctx, 5 * time.Second)
	defer cancel()
	req, err := http.NewRequestWithContext(reqCtx, http.MethodGet, versionCheckURL, nil)
	if err != nil {
		return "", fmt.Errorf("create GitHub request: %w", err)
	}

	resp, err := s.httpClient.Do(req)
```